File: 08-2-1-SimpleFactory/Game.cs
namespace SimpleFactory
{
    using System;

    public sealed class Game
    {
        private const int PipsOnDice = 6;

        public void Play()
        {
            var dice = new Dice(PipsOnDice);

            var roll1 = dice.Roll();
            var roll2 = dice.Roll();
            var sum = roll1 + roll2;
            var hasWon = sum == 2 * PipsOnDice;
            var message = hasWon ? "You win!" : "You loose!";

            Print(message);
        }

        private static void Print(string message) => Console.WriteLine(message);
    }
}

File: 08-DesignPatterns-Java/Solution/SimpleFactory/src/Game.java
import Dices.Dice;
import Factories.IDiceFactory;

public final class Game
{
    private static int PipsOnDice = 6;
    private final IDiceFactory _diceFactory;

    public Game(IDiceFactory diceFactory) {
        _diceFactory = diceFactory;
    }

    public void play()
    {
        var dice = _diceFactory.createDice(PipsOnDice);

        var roll1 = dice.roll();
        var roll2 = dice.roll();
        var sum = roll1 + roll2;
        var hasWon = sum == 2 * PipsOnDice;
        var message = hasWon ? "You win!" : "You loose!";

        print(message);
    }

    private static void print(String message) {
        System.out.println(message);
    }
}
